src/meta/src/hummock/manager/compaction.rs

```diff
@@ -265,6 +266,69 @@ impl HummockManager {
             } => {}
         }
     }
+
+    pub(crate) async fn calculate_vnode_partition(
+        &self,
+        compact_task: &mut CompactTask,
+        compaction_config: &CompactionConfig,
+    ) {
+        if compaction_config.split_weight_by_vnode > 0 {
+            for table_id in &compact_task.existing_table_ids {
+                compact_task
+                    .table_vnode_partition
+                    .insert(*table_id, compact_task.split_weight_by_vnode);
+            }
+        } else {
+            let mut table_size_info: HashMap<u32, u64> = HashMap::default();
+            let mut existing_table_ids: HashSet<u32> = HashSet::default();
+            for input_ssts in &compact_task.input_ssts {
+                for sst in &input_ssts.table_infos {
+                    existing_table_ids.extend(sst.table_ids.iter());
+                    for table_id in &sst.table_ids {
+                        *table_size_info.entry(*table_id).or_default() +=
+                            sst.file_size / (sst.table_ids.len() as u64);
+                    }
+                }
+            }
+            compact_task
+                .existing_table_ids
+                .retain(|table_id| existing_table_ids.contains(table_id));
+
+            let hybrid_vnode_count = self.env.opts.hybird_partition_vnode_count;
+            let default_partition_count = self.env.opts.partition_vnode_count;
+            let params = self.env.system_params_reader().await;
+            let barrier_interval_ms = params.barrier_interval_ms() as u64;
+            let checkpoint_secs = std::cmp::max(
+                1,
+                params.checkpoint_frequency() * barrier_interval_ms / 1000,
+            );
+            // check latest write throughput
+            let history_table_throughput = self.history_table_throughput.read();
+            for (table_id, compact_table_size) in table_size_info {
+                let write_throughput = history_table_throughput
+                    .get(&table_id)
+                    .map(|que| que.back().cloned().unwrap_or(0))
+                    .unwrap_or(0)
+                    / checkpoint_secs;
+                if compact_table_size > compaction_config.max_compaction_bytes / 2 {
+                    compact_task
+                        .table_vnode_partition
+                        .insert(table_id, default_partition_count);
```

Comment:
It took me a while to understand why we assign `default_partition_count` for table with size > `max_compaction_bytes / 2`. It is because we assign `default_partition_count` for the split compaction group and we want to treat the table in the hybrid group with large size in the task the same. This can easily confuse others. Can we add some comments?

Reply:
Just a magic number. I need some value to decide whether to partition for large task
But how to decide whether a task is large task ?
I can give another value which is `default_partition_count  * compaction_config.target_file_size_base`